### PR TITLE
feat: add attendee to event

### PR DESCRIPTION
**Pull request by Sofiag8**
## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics?
- [ ] Will this be part of a product update? If yes, please write one phrase about this update.

## Pull request type

Please check the type of change your PR introduces:

- [x] :sparkles: Feature
- [ ] :bug: Bugfix
- [ ] :pencil2: Fixing typos
- [ ] :lipstick: :art: Code style update (formatting, renaming)
- [x] :recycle: Refactoring (no functional changes, no api changes)
- [ ] :zap: Performance
- [ ] :pencil: Documentation content changes
- [x] :white_check_mark: Unit testing
- [x] :ok_hand: Updating due to code review suggest
- [ ] :rocket: Release: v
- [ ] :wrench: Add or update configuration files
- [ ] :hammer: Add or update development scripts
- [ ] :bricks: Infrastructure related changes

## What is the current module?

The endpoint existing at v1 about adding an user to an event is not working correctly and not good implemented

## What is the new behavior?

Now a v2 we will have the new endpoint for adding an user to an event with necessary validations 

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/modules/events/event.controller.ts [hunk 1]
```diff
@@ -38,4 +40,12 @@ export class EventController {
 
     return await this.eventService.getAll(params)
   }
+
+  @Post('/:eventId/attendee/:userId')
+  @HttpCode(200)
+  async addAttendeeToEvent(@Param() params: AddAttendeeToEventParams , @Body() attendeeData: AddAttendeeToEventDto): Promise<EventResponse> {
+    this.logger.log('Add attendee to event')
```
Comment (FelipeLosada1996) [inline src/modules/events/event.controller.ts:47]: In what help us that log? we can add some extra data to find what can break the endpoint or if we need what gonna be created

File: src/modules/events/event.service.ts [hunk 2]
```diff
@@ -41,4 +44,31 @@ export class EventService {
...
+
+  private validateEventStatus(eventId: ObjectId, event: Event) {
+    this.logger.log('Validating event status to add attendee...')
+    if (event.status !== EventStatus.ACTIVE) {
+      throw  new ChupitosBadRequestException(`Event with id: ${eventId} cannot accept attendees because is not active`)
+    }
+  }
+
+  private validateUserAttendingEvent(event: Event, eventId: ObjectId, userId: ObjectId) {
+    this.logger.log('Validating event attendees to avoid duplicated attendees...')
```
Comment (FelipeLosada1996) [inline src/modules/events/event.service.ts:69]: What exactly do this log? is help us to something?

**Comment by Sofiag8** [comment]
@FelipeLosada1996  I fixed the master conflicts, and also merge changes with the previous created mock files, I cleaned tests files avoiding repeated mocked code.

File: src/modules/events/event.controller.ts [hunk 3]
```diff
@@ -50,8 +53,16 @@ export class EventController {
 
   @Get('/:eventId')
   @HttpCode(200)
-  async getById(@Param('eventId') eventId: string): Promise<EventResponse> {
+  async getById(@Param('eventId') eventId: Types.ObjectId): Promise<EventResponse> {
```
Comment (FelipeLosada1996) [inline src/modules/events/event.controller.ts:56]: This is not a good practice in the controller, check  https://stackoverflow.com/questions/71597229/nestjs-validate-param-format-of-mongo-objectid

File: src/modules/events/event.controller.spec.ts [hunk 4]
```diff
@@ -71,43 +53,42 @@ describe('EventController', () => {
...
   describe('get by id', () => {
     it('should return the information about one event', async () => {
-      const id = '65b9373f6f3ef0a59e20975a'
-      const response: EventResponse = EVENTMOCK
-      jest.spyOn(service, 'getById').mockResolvedValue(response)
+      const id = EVENT_ID_MOCK
+      jest.spyOn(service, 'getById').mockResolvedValue(EVENT_RESPONSE_MOCK)
 
-      const result = await controller.getById(id)
+      const result = await controller.getById(Object(id))
```
Comment (FelipeLosada1996) [inline src/modules/events/event.controller.spec.ts:89]: What is the difference with `Object(EVENT_ID_MOCK)`?

File: src/modules/events/event.controller.spec.ts [hunk 5]
```diff
@@ -71,43 +53,43 @@ describe('EventController', () => {
...
 
   describe('get by id', () => {
     it('should return the information about one event', async () => {
-      const id = '65b9373f6f3ef0a59e20975a'
-      const response: EventResponse = EVENTMOCK
+      const id = EVENT_ID_MOCK
+      const response: EventResponse = EVENT_RESPONSE_MOCK
       jest.spyOn(service, 'getById').mockResolvedValue(response)
 
       const result = await controller.getById(id)
```
[the remark this answers is not in the document]
Comment (Sofiag8) [reply src/modules/events/event.controller.spec.ts]: done

File: src/modules/events/event.controller.spec.ts (hunk 4 above)
Comment (Sofiag8) [reply src/modules/events/event.controller.spec.ts]: done

**Review by kellyCSalazar — APPROVED** [review]

**Review by FelipeLosada1996 — APPROVED** [review]

**Pull request merged**